src/PaymentGateways/Stripe/Admin/AccountManagerSettingField.php

```diff
@@ -0,0 +1,564 @@
+<?php
+
+namespace Give\PaymentGateways\Stripe\Admin;
+
+use Give\PaymentGateways\Stripe\Repositories\AccountDetail;
+use Give\PaymentGateways\Stripe\Repositories\Settings;
+use Give_Admin_Settings;
+use function _e;
+use function add_query_arg;
+use function do_action;
+use function esc_html;
+use function esc_html__;
+use function esc_html_e;
+use function esc_url_raw;
+use function get_site_url;
+use function give;
+use function esc_attr;
+use function give_has_upgrade_completed;
+use function give_stripe_connection_type_name;
+use function give_stripe_is_premium_active;
+
+/**
+ * Class AccountManagerSettingField
+ *
+ * @package Give\PaymentGateways\Stripe\Admin
+ * @unreleased
+ */
+class AccountManagerSettingField {
+
+	/**
+	 * @var AccountDetail
+	 */
+	private $accountDetailRepository;
+
+	/**
+	 * @var array
+	 */
+	private $stripeAccounts;
+
+	/**
+	 * @var string
+	 */
+	private $defaultStripeAccountSlug;
+	/**
+	 * @var Settings
+	 */
+	private $settings;
+
+	/**
+	 * AccountManagerSettingField constructor.
+	 *
+	 * @unreleased
+	 *
+	 * @param AccountDetail $accountDetailRepository
+	 * @param Settings $settings
+	 */
+	public function __construct( AccountDetail $accountDetailRepository, Settings $settings ) {
+		$this->accountDetailRepository = $accountDetailRepository;
+		$this->settings = $settings;
+	}
+
+	/**
+	 * @unreleased
+	 */
+	private function setUpProperties() {
+		global $post;
+		$this->stripeAccounts           = $this->settings->getAllStripeAccounts();
+		$this->defaultStripeAccountSlug = $this->isGlobalSettingPage() ?
+			$this->settings->getDefaultStripeAccountSlug() :
+			$this->settings->getDefaultStripeAccountSlugForDonationForm( $post->ID );
+	}
+
+	/**
+	 * Render Stripe account manager setting field.
+	 *
+	 * @unreleased
+	 *
+	 * @param array $field
+	 */
+	public function handle( $field ) {
+		$this->setUpProperties();
+		$classes = ! empty( $field['wrapper_class'] ) ? esc_attr( $field['wrapper_class'] ) : ''
+		?>
+		<div class="<?php echo $classes; ?>">
+
+			<div id="give-stripe-account-manager-errors"></div>
+			<?php $this->getIntroductionSectionMarkup(); ?>
+			<div class="give-stripe-account-manager-container">
+				<div class="main-heading">
+					<h2 class="give-stripe-setting-heading"><?php esc_html_e( 'Connected Accounts', 'give' ); ?></h2>
+				</div>
+				<?php
+				$this->getStripeAccountListSectionMarkup();
+				$this->getAddNewStripeAccountSectionMarkup();
+				?>
+			</div>
+			<?php $this->getDefaultStripeAccountNotice(); ?>
+		</div>
+		<?php
+	}
+
+	/**
+	 * @unreleased
+	 */
+	private function getIntroductionSectionMarkup() {
+		// Show introduction content only on global setting edit screen.
+		if ( ! $this->isGlobalSettingPage() ) {
+			return;
+		}
+		?>
+		<div id="give-stripe-account-manager-description">
```

Comment:
Probably not worth it by this point, but I encourage you to start making more use of views when building and refactoring. That will really clean up a lot of this sort of code.

Reply:
@JasonTheAdams Noted.